Scripts/GetFieldsByIncidentType/GetFieldsByIncidentType.py

```diff
@@ -0,0 +1,79 @@
+import demistomock as demisto
+from CommonServerPython import *
+from pprint import pformat
+
+
+def get_field_by_long_name(fields, name):
+    for field in fields:
+        if field['name'] == name:
+            return field
+    return None
+
+
+def get_field_by_short_name(fields, name):
+    for field in fields:
+        if field['cliName'] == name:
+            return field
+    return None
+
+
+def main():
+    # get incident fields
+    res = demisto.executeCommand('demisto-api-get', {'uri': '/incidentfields'})
+    if is_error(res):
+        return_error(res[0]['Contents'])
+
+    fields = res[0]['Contents']['response']
+
+    # 'fields' contains non-incident fields, as well, so let's make a version containing only incident fields
+    incident_fields = [field for field in fields if field['id'].startswith('incident_')]
+
+    # get arguments
+    args = demisto.args()
+
+    incident_type = args['incident_type']
+
+    exclude_system = False
+    if 'custom' in args and argToBoolean(args['custom']) is True:
+        exclude_system = True
+
+    name_key = 'name'
+    if 'short_names' in args and argToBoolean(args['short_names']) is True:
+        name_key = 'cliName'
+
+    explicit_only = False
+    if 'explicit_only' in args and argToBoolean(args['explicit_only']) is True:
+        explicit_only = True
+
+    # generate results
+    types = []
+    if exclude_system is True:
+        # only return non-system fields
+        for field in incident_fields:  # using multiple if statements for readability
+            if field['system'] is False:  # exclude system fields
+                if field['associatedToAll'] is True and explicit_only is False:
+                    # if explicit_only is false, include fields associated to all incident types
+                    types.append(field[name_key])
+                elif field['associatedTypes'] is not None and incident_type in field['associatedTypes']:
```

Comment:
When I'm looking for a null value, I like to explicitly say that's what I'm looking for, as opposed to other truthy values, as it says something about what's being evaluated.

Reply:
A reason I use the 'is True|False' is that since this is referring to dict property, a string or numeric value other than zero would also evaluate true, which I don't want.  Using `is True` is more appropriate as it enforces boolean type checking on the property.

This is also why I like using `is True|False` rather than `if x` or `if not x` -- I want to fail right away if a non-boolean type value somehow creeps its way into the data.  It's also a micro-optimisation in that all it has to do, in theory, is a reference comparison.

To sum it up, I like type checking.